Improve Usability of the SANS Beam Centre Finder

cailafinn opened the pull request:
### Description of work

#### Summary of work
Improve the usability of a few features in the SANS Beam Centre Finder. 

Fixes #36518 
Fixes #36435 

**Report to:** @smk78 

#### Further detail of work
1. LARMOR sets it's beam centres on the GUI in meters (in comparison to the other instruments, which use mm), so the GUI now automatically updates the labels to indicate the correct unit. 
2. The confusing check box and combobox combination has now been replaced by a pair of radio buttons for selecting which of the two banks the beam centre should be found for. 
3. Both the front and rear values are loaded into the GUI from the user file when the reduction mode is set to `front` or `rear`. The behaviour for `all` and `merged` is unchanged. 

### To test:
1. Open the SANS GUI and load the user file from the `loqdemo` in the Training Course Data
    1. In this user file, the `selected_detector` should be set to `front`.
1. The values for both front and rear should have been loaded in from the user file into the beam centre finder. 
2. Check that the radio button has automatically selected the Find Front option. 
3. Run the beam centre finder and check that the output value (in the log on the right) matches the one set in the BCF's text boxes. 
4. Edit the user file to change the instrument from `LOQ` to `LARMOR` 
5. Reload the user file and navigate back to the BCF. The labels above the coordinate boxes should now be in meters rather than mm. 

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

Commit:
Fill both front and rear when loading

Even if the value isn't supposed to be used in the reduction, the
suggestion from Steve was that both values are filled in when front/rear
is selected.

RE #36518

Commit:
Allow changing of BCF units to meters

This only affects the GUI. The underlying code was just doing this in
the background anyway.

RE #36518

Commit:
Change BCF labels when LARMOR user file loaded

RE #36518

Commit:
Improve clarity of BCF bank selection

RE #36518

Commit:
Add a release note

RE #36518

rbauststfc commented on scripts/Interface/ui/sans_isis/beam_centre.py:
```diff
@@ -148,10 +147,18 @@ def set_run_button_to_normal(self):
-        self.update_rear_check_box.setChecked(enabled)
+        self.update_rear_radio.setChecked(enabled)
+
+    def set_scale_millimeters(self):
```
Might it be worth reducing a bit of duplication here by consolidating these two methods into something like the below?

```
def set_scale(self, is_millimeters=True):
    if is_millimeters:
        scale = "mm"
    else:
        scale = "m"
    self.rear_centre_label.setText(f"Centre Position - Rear ({scale})")
    self.front_centre_label.setText(f"Centre Position - Front ({scale})")
```

rbauststfc commented on scripts/SANS/sans/gui_logic/presenter/beam_centre_presenter.py:
```diff
@@ -128,6 +130,12 @@ def copy_centre_positions(self, state_model):
         self._beam_centre_model.front_pos_1 = getattr(state_model, "front_pos_1")
         self._beam_centre_model.front_pos_2 = getattr(state_model, "front_pos_2")
 
+    def set_scale_to_meters(self, enable: bool) -> None:
```
Very minor suggestion, but would it be clearer to rename this to `set_scale`, just to make it a bit clearer from the name that it's not always going to set the scale to meters? The boolean parameter could then potentially be re-named to `is_meters`.

rbauststfc reviewed (commented):
Thanks for this, it makes the interface much more intuitive to work with. The code changes look good - I've made a minor suggestion that you can feel free to take or leave, it's just a thought.

On testing, the changes worked well until I changed the training data TOML file to be instrument LARMOR with selected detector set to Rear. With these settings I was unable to load the mask file and got error: "The following key is missing: 'HAB'". I tried to double check in the Nightly and this combination of settings seem to have worked before, so this might need a look.

Commit:
Improve methods to remove code duplication

RE #36518

Commit:
Improve handling of incorrect detector positions

If a TOML file is containing centre positions for detectors that don't
exist, we should not be attempting to assign them to a non-existant
detectors. Instead, only assign the centre positions for detectors that
are valid for the selected instrument.

RE #36518

Commit:
Add a test for gracefully handling front on single bank instruments

RE #36518

rbauststfc reviewed (approved):
Code changes all look nice and clean with good test coverage. The functionality is all working well and the issue with loading some mask file settings for instruments with only a rear detector has now been resolved (and an appropriate test added for the bug). I've tested lots of different combinations of mask file detector configuration settings for both LOQ and LARMOR and it all seems to be working well. All review comments have been addressed and this looks ready to be merged in to me - nice work. 👍
